src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -100,28 +127,66 @@ export class CodeWhispererCodeCoverageTracker {
             } catch (e) {
                 getLogger().verbose(`Exception Thrown from CodeWhispererCodeCoverageTracker: ${e}`)
             } finally {
-                this._totalTokens = []
-                this._acceptedTokens = []
+                this._totalTokens = {}
+                this._acceptedTokens = {}
                 this._startTime = 0
                 this.closeTimer()
             }
         }, CodeWhispererConstants.defaultCheckPeriodMillis)
     }
 
-    public closeTimer() {
+    private closeTimer() {
         if (this._timer !== undefined) {
             clearTimeout(this._timer)
             this._timer = undefined
         }
     }
 
-    public static readonly instances = new Map<telemetry.CodewhispererLanguage, CodeWhispererCodeCoverageTracker>()
-    public static getTracker(
-        language: telemetry.CodewhispererLanguage = 'plaintext',
-        memento: vscode.Memento
-    ): CodeWhispererCodeCoverageTracker {
-        const instance = this.instances.get(language) ?? new this(language, memento)
-        this.instances.set(language, instance)
-        return instance
+    public addAcceptedTokens(filename: string, token: CodeWhispererToken) {
+        if (!(filename in this._acceptedTokens)) {
+            this._acceptedTokens[filename] = []
+        }
+        this._acceptedTokens[filename].push(token)
+    }
+
+    public addTotalTokens(filename: string, count: number) {
+        if (!(filename in this._totalTokens)) {
+            this._totalTokens[filename] = 0
+        }
+        this._totalTokens[filename] += count
+        if (this._totalTokens[filename] < 0) {
+            this._totalTokens[filename] = 0
+        }
+    }
+
+    public countTotalTokens(e: vscode.TextDocumentChangeEvent) {
+        // ignore no contentChanges. ignore contentChanges from other plugins (formatters)
+        // only include contentChanges from user action
+        if (
+            !CodeWhispererConstants.supportedLanguages.includes(e.document.languageId) ||
+            vsCodeState.isCodeWhispererEditing ||
+            e.contentChanges.length !== 1
+        )
+            return
+        const content = e.contentChanges[0]
+        // do not count user tokens if user copies large chunk of code
+        if (content.text.length > 20) return
+        this.tryStartTimer()
+        if (content.text.length === 0) {
+            this.addTotalTokens(e.document.fileName, -content.rangeLength)
```

Comment:
nit: It took me a while to figure out this is the delete case, could you pls add a comment here?